src/lib/lang-runner-java.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as os from 'os';
+import * as path from 'path';
+import * as fs from 'fs';
+import * as https from 'https';
+import * as execa from 'execa';
+import LangRunner from '../lib/lang-runner';
+import LocalRun from '../lib/local-run';
+
+const runtimeJarName = 'sf-fx-runtime-java-runtime-1.0.3-jar-with-dependencies.jar';
+const runtimeJarUrl = `https://repo1.maven.org/maven2/com/salesforce/functions/sf-fx-runtime-java-runtime/1.0.3/${runtimeJarName}`;
+const runtimeJarDir = path.resolve(os.tmpdir(), 'sf-fx-runtime-java-runtime-jar');
+const runtimeJarPath = path.resolve(runtimeJarDir, runtimeJarName);
+const runtimeJarSha = '1db6d78bdbb7aff7ebe011565190ca9dd4d3e68730e206628230d480d057fe1e';
+
+export default class LangRunnerJava extends LangRunner {
+  async detect(): Promise<boolean> {
+    const pomXmlPath = path.resolve(this.path, 'pom.xml');
+    try {
+      await fs.promises.access(pomXmlPath);
+    } catch {
+      return false;
+    }
+    return true;
+  }
+
+  async build(): Promise<void> {
+    await this.checkJava();
+    await this.ensureRuntimeJar();
+    await this.runMavenInstall();
+  }
+
+  async start(): Promise<void> {
+    await this.runRuntimeJarServe();
+  }
+
+  private async checkJava(): Promise<void> {
+    try {
+      await execa.command('java --version');
+    } catch (error) {
+      throw new Error('Java executable not found.');
+    }
+  }
+
+  private async ensureRuntimeJar(): Promise<void> {
+    if (await this.checkRuntimeJar()) {
+      return;
+    }
+    await this.downloadRuntimeJar();
+  }
+
+  private async runMavenInstall(): Promise<void> {
+    try {
+      await execa.command(`${path.resolve(this.path, 'mvnw')} install`, { stdio: 'inherit', cwd: this.path });
```

Comment:
This should use the `compile` goal instead of `install`. Installing will write the resulting artefact to the users local Maven cache which takes some extra time and might be unexpected. Everything that is required to run the function will be executed by `compile`.

In addition, we need to support the case where `mvnw` is not present in the users function project. Even though we encourage users to use Maven wrapper and the template creates a project with it, we support functions without Maven wrapper everywhere else.

Reply:
Fixed in https://github.com/salesforcecli/plugin-functions/pull/260/commits/ab2bfc810c93049c0979998802171aedec3db26b. Good feedback. Thanks!